extractor/youtube/core.py

```diff
@@ -1,10 +1,48 @@
 import requests
 from bs4 import BeautifulSoup
 from youtube_title_parse import get_artist_title
+from helpers import get_yt_link, clean_channel
 
-def scrape_yt(url):
-    response = requests.get(url)
-    soup = BeautifulSoup(response.content, 'lxml')
+# Scrapper function for videos without "Music in this video" section
+def scrape_yt(soup):
     raw_title = soup.find('meta', {'property': 'og:title'}).get('content').strip()
-    artist, title = raw_title.split(' - ')
-    return artist, title
+    artist,title = None,None
+
+    #In case the YouTube Title is in good format,i.e, Artist-Song("foo")
+    if '-' in raw_title:
+        raw_title =  get_artist_title(raw_title)
+        artist, title = raw_title.split(' - ')
+
+    # In case the  YouTube Title only contains song name    
+    else :
+        title = get_artist_title(raw_title)
+        artist = soup.find('a',class_= "yt-uix-sessionlink spf-link").text
```

Comment:
Can you explain (as a reply to this) what this line is doing? I'm not sure I understand why we're treating titles with `-` separately.

Reply:
Most of the youtube music video titles have format as "Artist - Song". So I used it.

Reply:
Corrected it.